test/e2e/daemon_set.go

```diff
@@ -399,44 +395,112 @@ var _ = framework.KubeDescribe("Daemon set [Serial]", func() {
 		newDS, err = c.Extensions().DaemonSets(ns).Create(newDS)
 		Expect(err).NotTo(HaveOccurred())
 		Expect(newDS.Spec.TemplateGeneration).To(Equal(templateGeneration))
+		Expect(apiequality.Semantic.DeepEqual(newDS.Spec.Template, ds.Spec.Template)).To(BeTrue(), "DaemonSet template should match to adopt pods")
 
-		By(fmt.Sprintf("Wait for all pods to be adopted by DaemonSet %s", newDSName))
-		err = wait.PollImmediate(dsRetryPeriod, dsRetryTimeout, checkDaemonSetPodsAdopted(c, ns, newDS.UID, label))
-		Expect(err).NotTo(HaveOccurred(), "error waiting for DaemonSet pods to be orphaned")
+		By(fmt.Sprintf("Wait for pods and history to be adopted by DaemonSet %s", newDS.Name))
```

Comment:
Removed all `By` in these 2 e2e tests. Kept other tests as-is to make commit cleaner. We can remove them in follow up PRs.